Source/Scene/ImplicitSubtree.js

```diff
@@ -0,0 +1,470 @@
+ * @returns {Object} A dictionary of buffer view index to a Uint8Array of its contents.
+ * @private
+ */
+function parseActiveBufferViews(bufferViewHeaders, buffersU8) {
+  var bufferViewsU8 = {};
+  for (var i = 0; i < bufferViewHeaders.length; i++) {
+    var bufferViewHeader = bufferViewHeaders[i];
+
+    if (!bufferViewHeader.isActive) {
+      continue;
+    }
+
+    var start = bufferViewHeader.byteOffset;
+    var end = start + bufferViewHeader.byteLength;
+    var buffer = buffersU8[bufferViewHeader.buffer];
+    var bufferView = buffer.subarray(start, end);
+    bufferViewsU8[i] = bufferView;
+  }
+  return bufferViewsU8;
+}
+
+/**
+ * Parse the three availability bitstreams and store them in the subtree
+ *
+ * @param {ImplicitSubtree} subtree The subtree to modify
+ * @param {Object} subtreeJson The subtree JSON
+ * @param {ImplicitTileset} implicitTileset The implicit tileset this subtree belongs to
+ * @param {Object} bufferViewsU8 A dictionary of buffer view index to a Uint8Array of its contents.
+ * @private
+ */
+function parseAvailability(
+  subtree,
+  subtreeJson,
+  implicitTileset,
+  bufferViewsU8
+) {
+  var branchingFactor = implicitTileset.branchingFactor;
+  var subtreeLevels = implicitTileset.subtreeLevels;
+  var tileAvailabilityBits =
+    (Math.pow(branchingFactor, subtreeLevels) - 1) / (branchingFactor - 1);
+  var childSubtreeBits = Math.pow(branchingFactor, subtreeLevels);
+
+  subtree._tileAvailability = parseAvailabilityBitstream(
+    subtreeJson.tileAvailability,
+    bufferViewsU8,
+    tileAvailabilityBits
+  );
+
+  subtree._contentAvailability = parseAvailabilityBitstream(
+    subtreeJson.contentAvailability,
+    bufferViewsU8,
+    // content availability has the same length as tile availability.
+    tileAvailabilityBits
+  );
+  subtree._childSubtreeAvailability = parseAvailabilityBitstream(
+    subtreeJson.childSubtreeAvailability,
+    bufferViewsU8,
+    childSubtreeBits
+  );
+}
+
+/**
+ * Given the JSON describing an availability bitstream, turn it into an
+ * in-memory representation using an {@link ImplicitAvailabilityBitstream}
+ * object. This handles both constants and bitstreams from a bufferView.
+ *
+ * @param {Object} availabilityJson A JSON object representing the availability
+ * @param {Object} bufferViewsU8 A dictionary of bufferView index to its Uint8Array contents.
+ * @param {Number} lengthBits The length of the availability bitstream in bits
+ * @returns {ImplicitAvailabilityBitstream} The parsed bitstream object
+ * @private
+ */
+function parseAvailabilityBitstream(
+  availabilityJson,
+  bufferViewsU8,
+  lengthBits
+) {
+  if (defined(availabilityJson.constant)) {
+    return new ImplicitAvailabilityBitstream({
+      constant: Boolean(availabilityJson.constant),
+      lengthBits: lengthBits,
+      availableCount: availabilityJson.availableCount,
+    });
+  }
+
+  if (defined(availabilityJson.bufferView)) {
```

Comment:
Shouldn't either `constant` or `bufferView` be defined according to the spec? This if statement might not be necessary.